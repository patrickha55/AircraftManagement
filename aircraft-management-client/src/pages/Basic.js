import React from 'react'

const Basic = (props) => {
    return (
        <div>
            {
                props.name && <p>Hello {props.name}</p>
            }
            {
                !props.name && <p>Welcome</p>
            }
        </div>
    )
}

export default Basic;