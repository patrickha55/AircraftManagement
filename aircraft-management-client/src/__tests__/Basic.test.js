import renderer from 'react-test-renderer';
import Basic from '../pages/Basic';
import React from 'react';

test("Basic page should renders with or without name", () => {
    //arrange
    let tree = renderer
        .create(<Basic />)
        .toJSON();

    //assert
    expect(tree).toMatchSnapshot();

    tree = renderer
        .create(<Basic name="Phat" />)
        .toJSON();

    //assert
    expect(tree).toMatchSnapshot();
})