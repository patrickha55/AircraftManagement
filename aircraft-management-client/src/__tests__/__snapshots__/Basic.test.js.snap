// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic page should renders with or without name 1`] = `
<div>
  <p>
    Welcome
  </p>
</div>
`;

exports[`Basic page should renders with or without name 2`] = `
<div>
  <p>
    Hello 
    Phat
  </p>
</div>
`;
